File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/Reports/ReportBuilderPropertyModel.ts
import Environment = require('App/Models/Environment');
import ReportBuilderItem = require('App/Models/Reports/ReportBuilderItem');
import ReportBuilderPropertyType = require('App/Models/Reports/ReportBuilderPropertyType');

class ReportBuilderPropertyModel implements ReportBuilderItem {
    constructor() {
        this.Children = [];
    }

    Name: string;
    DisplayName: string;
    PropertyType: ReportBuilderPropertyType;
    EntityType: Environment;
    Children: ReportBuilderPropertyModel[];
    get HasChildren() { return this.Children.length > 0; }
}

export = ReportBuilderPropertyModel;

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Reports/Implementation/ReportBuilderPropertyModel.cs
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Model.Entities.Commons;

namespace VecompSoftware.DocSuiteWeb.Model.Reports
{
    public class ReportBuilderPropertyModel : IReportItem
    {
        #region [ Constructor ]
        public ReportBuilderPropertyModel()
        {
            Children = new List<ReportBuilderPropertyModel>();
        }
        #endregion

        #region [ Properties ]
        public string Name { get; set; }
        public string DisplayName { get; set; }
        public ReportBuilderPropertyType PropertyType { get; set; }
        public ReportBuilderPropertyCondition Condition { get; set; }
        public DSWEnvironmentType? Environment { get; set; }
        public ICollection<ReportBuilderPropertyModel> Children { get; set; }
        public bool HasChildren => Children.Count > 0;
        #endregion        
    }
}
